Plone5 compat

alecpm opened the pull request:
Would be great of @witekdev and/or @davisagli could review these changes for Plone 5 compatibility.  I haven't been able to confirm if this maintains Plone 3.x compatibility, but it should work with Plone 4.x, including 4.3 using new-style collections instead of ATTopics when installed.  This still uses AT for the content types, but under Plone 5 will use new mockup-based AT widgets, new dexterity collections, new resource registry/bundles, updated registry config for TinyMCE, etc.

alecpm committed:
Initial stab at Plone 5 compatibility.

alecpm committed:
Test updates, fixes.

alecpm committed:
Add support for new-style collections in Plone 4.x and test rig for Plone 4.3

witekdev commented:
@alecpm Thanks! I'll have a look at this soon and report back..

witekdev commented:
@alecm
Here are my brief initial findings. On Plone 5.0.5 it installs cleanly.

It fails to uninstall, giving:
`2016-07-29 06:30:23 ERROR root Could not uninstall Products.PressRoom: undefined property 'default_page_types'`

![screenshot from 2016-07-29 06 32 40](https://cloud.githubusercontent.com/assets/883619/17248479/6c92ecda-5560-11e6-87f9-adb233b76b3f.png)

Improvements/nice to have:
It would be good to provide a toolbar menu icon (even if it is just the same one for all content types) right now we have blank icons.

![screenshot from 2016-07-29 06 32 52](https://cloud.githubusercontent.com/assets/883619/17248475/63236bac-5560-11e6-9022-90dc79f0f2bb.png)

alecm commented:
@witekdev -  you might want to notify @alecpm - I've nothing to do w/ this project :-)

witekdev commented:
@alecm :)
sorry about that :)

witekdev commented:
@alecpm see above. thanks!

witekdev commented:
@alecpm I updated my comment above, things went a lot smoother with _Archetypes Content Types for Plone_ enabled. I was able to create all 4 content types without issues so far.
Folder contents/circular reference issue is solved by pulling in our version of
plone.app.content = git https://github.com/jazkarta/plone.app.content.git

alecpm committed:
Plone 5 fixes.

alecpm commented:
Looks like it needs the Products.ATContentyTypes:base profile installed for the factory tool and other things. I've added that. I also added some css for add menu icons, but the resource registry stuff isn't working for me unless the bundle CSS is in development mode. Not sure what's going on there, but I think it's pretty clearly a bug in the Plone RR.

alecpm committed:
Template compatilibity fixes for Plone 5.1

cguardia committed:
remove calls to no-longer-existing method

cguardia committed:
fix add links for pressroom view

alecpm committed:
Remove reference browser dependency completely, update pattern defaults.

cguardia committed:
fix press cl;ips and press releases links

witekdev committed:
Fix for plone.batching.batch IndexError: slice

witekdev committed:
Handle case where number of PressReleases or PressClips is not set, display all in that case

alecpm committed:
Fix error in pressroom listing view.
